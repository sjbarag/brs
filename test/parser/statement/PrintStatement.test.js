const { Lexeme, Parser } = require("brs");
const BrsError = require("../../../lib/Error");

const { token, EOF } = require("../ParserTests");

describe("parser", () => {
    afterEach(() => BrsError.reset());

    describe("print statements", () => {
        it("parses singular print statements", () => {
            let parsed = Parser.parse([
                token(Lexeme.Print),
                token(Lexeme.String, "Hello, world"),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses print lists with no separator", () => {
            let parsed = Parser.parse([
                token(Lexeme.Print),
                token(Lexeme.String, "Foo"),
                token(Lexeme.String, "bar"),
                token(Lexeme.String, "baz"),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses print lists with separators", () => {
            let parsed = Parser.parse([
                token(Lexeme.Print),
                token(Lexeme.String, "Foo"),
                token(Lexeme.Semicolon),
                token(Lexeme.String, "bar"),
                token(Lexeme.Comma),
                token(Lexeme.String, "baz"),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });
});
