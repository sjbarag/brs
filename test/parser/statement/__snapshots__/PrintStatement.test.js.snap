// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser print statements parses print lists with no separator 1`] = `
Array [
  Print {
    "expressions": Array [
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "Foo",
        },
      },
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "bar",
        },
      },
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "baz",
        },
      },
    ],
    "tokens": Object {
      "print": Object {
        "isReserved": false,
        "kind": "Print",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": undefined,
      },
    },
    "type": "Print",
  },
]
`;

exports[`parser print statements parses print lists with separators 1`] = `
Array [
  Print {
    "expressions": Array [
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "Foo",
        },
      },
      Object {
        "isReserved": false,
        "kind": "Semicolon",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": undefined,
      },
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "bar",
        },
      },
      Object {
        "isReserved": false,
        "kind": "Semicolon",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": undefined,
      },
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "baz",
        },
      },
    ],
    "tokens": Object {
      "print": Object {
        "isReserved": false,
        "kind": "Print",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": undefined,
      },
    },
    "type": "Print",
  },
]
`;

exports[`parser print statements parses singular print statements 1`] = `
Array [
  Print {
    "expressions": Array [
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": undefined,
      },
    ],
    "tokens": Object {
      "print": Object {
        "isReserved": false,
        "kind": "Print",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": undefined,
      },
    },
    "type": "Print",
  },
]
`;

exports[`parser print statements supports empty print 1`] = `
Array [
  Print {
    "expressions": Array [
      Literal {
        "_location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "type": "Literal",
        "value": BrsString {
          "kind": 3,
          "value": "",
        },
      },
    ],
    "tokens": Object {
      "print": Object {
        "isReserved": false,
        "kind": "Print",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": undefined,
      },
    },
    "type": "Print",
  },
]
`;
